testing/simapp/simd/cmd/root.go

```diff
@@ -30,18 +28,24 @@ import (
 
 	dbm "github.com/cometbft/cometbft-db"
 	tmcfg "github.com/cometbft/cometbft/config"
-	tmcli "github.com/cometbft/cometbft/libs/cli"
 	"github.com/cometbft/cometbft/libs/log"
-	tmtypes "github.com/cometbft/cometbft/types"
 
 	"github.com/cosmos/ibc-go/v7/testing/simapp"
 	"github.com/cosmos/ibc-go/v7/testing/simapp/params"
 )
 
 // NewRootCmd creates a new root command for simd. It is called once in the
 // main function.
-func NewRootCmd() (*cobra.Command, params.EncodingConfig) {
-	encodingConfig := simapp.MakeTestEncodingConfig()
+func NewRootCmd() *cobra.Command {
+	// we "pre"-instantiate the application for getting the injected/configured encoding configuration
+	tempApp := simapp.NewSimApp(log.NewNopLogger(), dbm.NewMemDB(), nil, true, simtestutil.NewAppOptionsWithFlagHome(simapp.DefaultNodeHome))
+	encodingConfig := params.EncodingConfig{
+		InterfaceRegistry: tempApp.InterfaceRegistry(),
+		Marshaler:         tempApp.AppCodec(),
+		TxConfig:          tempApp.TxConfig(),
+		Amino:             tempApp.LegacyAmino(),
+	}
```

Comment:
Is this what the sdk simapp does? I thought we used to pass encoding config as an arg to `NewSimApp()`?

It looks like `MakeTestEncodingConfig()` used above is no longer used at all, so could be removed? Alternatively we could rename to `MakeEncodingConfig()` (keeping it public) and leverage that in the new app constructor, then we could also avoid this "pre-instantiation", right?

Also happy to keep it like this if this is the standard, but it feels a little off to me

Reply:
Aha, I see this is closely related to app_v2 wiring and is likely just a workaround until fully migrated to app v2 and depinject.

https://github.com/cosmos/cosmos-sdk/blob/main/simapp/simd/cmd/root.go#L49

Reply:
yes, this